torch/_tensor.py

```diff
@@ -1717,23 +1717,37 @@ def __dlpack__(self, stream=None, max_version=None):
             # Stream pointers in CUDA/ROCm are uniquely numbered and can
             # be retrieved from their integer value.
             raise TypeError("stream must be ``int`` or ``none``")
-        elif stream is not None and stream != -1:
+        elif stream != -1:
             if self.device.type == "cuda":
                 # NB: This logic handles the special case values for default
```

Comment:
No update to dlpack.py ? :D

Reply:
No need. If `stream` is None, we still need to synchronize, assuming the legacy default stream.